File: DaiPhucVinh/Application/src/api/unitOfMeasure/unitOfMeasureService.js
import { Proxy } from "../Proxy";
import { ServerEndPoint } from "../ServerEndPoint";
import { BaseResponse } from "../BaseResponse";
const ServiceEndPoint = ServerEndPoint + "/unitofmeasure";
const API = {
  TakeAllUnitOfMeasure: "/TakeAllUnitOfMeasure",
  TakeUnitOfMeasure: "/TakeUnitOfMeasure",
};

const TakeAllUnitOfMeasure = async (request) => {
  let result = new BaseResponse(false, "", null);
  try {
    let response = await Proxy(
      "post",
      ServiceEndPoint + API.TakeAllUnitOfMeasure,
      request,
      true
    );
    if (response.data.Success) {
      result.success = true;
      result.data = response.data.Data;
      result.dataCount = response.data.DataCount;
    }
  } catch (e) {
    result.message = e.toString();
  }
  return result;
};
const TakeUnitOfMeasure = async (Code) => {
  let result = new BaseResponse(false, "", null);
  try {
    let response = await Proxy(
      "get",
      ServiceEndPoint + API.TakeUnitOfMeasure,
      { params: { Code: Code } },
      true
    );
    if (response.Success) {
      result.success = true;
      result.Item = response.Item;
      result.dataCount = response.data.DataCount;
    }
  } catch (e) {
    result.message = e.toString();
  }
  return result;
};
export { TakeAllUnitOfMeasure, TakeUnitOfMeasure };

File: DaiPhucVinh/Services/MainServices/UnitOfMeasure/UnitOfMeasureService.cs
using DaiPhucVinh.Services.Database;
using DaiPhucVinh.Services.Framework;
using DaiPhucVinh.Services.Helper;
using DaiPhucVinh.Shared.Common;
using DaiPhucVinh.Shared.UnitOfMeasure;
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DaiPhucVinh.Services.MainServices.UnitOfMeasure
{
    public interface IUnitOfMeasureService
    {
        Task<BaseResponse<UnitOfMeasureResponse>> TakeAllUnitOfMeasure(UnitOfMeasureRequest request);
        Task<BaseResponse<UnitOfMeasureResponse>> TakeUnitOfMeasure(string Code);
    }
    public class UnitOfMeasureService : IUnitOfMeasureService
    {
        private readonly DataContext _datacontext;
        private readonly ILogService _logService;
        public UnitOfMeasureService(DataContext datacontext, ILogService logService)
        {
            _datacontext = datacontext;
            _logService = logService;
        }
        public async Task<BaseResponse<UnitOfMeasureResponse>> TakeAllUnitOfMeasure(UnitOfMeasureRequest request)
        {
            var result = new BaseResponse<UnitOfMeasureResponse> { };
            try
            {
                var query = _datacontext.WMS_UnitOfMeasures.AsQueryable();
                result.DataCount = await query.CountAsync();
                var data = await query.ToListAsync();
                result.Data = data.MapTo<UnitOfMeasureResponse>();
                result.Success = true;
            }
            catch (Exception ex)
            {
                result.Message = ex.ToString();
                _logService.InsertLog(ex);
            }
            return result;
        }
        public async Task<BaseResponse<UnitOfMeasureResponse>> TakeUnitOfMeasure(string Code)
        {
            var result = new BaseResponse<UnitOfMeasureResponse> { };
            try
            {
                var query = await _datacontext.WMS_Items.FirstOrDefaultAsync(x => x.Code == Code);
                result.Item = query.MapTo<UnitOfMeasureResponse>();
                result.Success = true;
            }
            catch (Exception ex)
            {
                result.Message = ex.ToString();
                _logService.InsertLog(ex);
            }
            return result;
        }
    }
}
